src/lib/Libtpp/tpp_transport.c

```diff
@@ -505,6 +505,19 @@ tpp_transport_init(struct tpp_config *conf)
 
 	if (tpp_init_rwlock(&cons_array_lock))
 		return -1;
+	
+	if (pthread_mutex_init(&tpp_nslookup_mutex, NULL) != 0) {
+		tpp_log(LOG_CRIT, __func__, "Failed to initialize nslookup mutex");
+		return -1;
+	}
+	
+#ifndef WIN32
+	/* for unix, set a pthread_atfork handler */
+	if (pthread_atfork(tpp_nslookup_atfork_prepare, tpp_nslookup_atfork_parent, tpp_nslookup_atfork_child) != 0) {
+		fprintf(stderr, "tpp nslookup mutex atfork handler failed\n");
+		exit(1);
```

Comment:
Is there any need to exit here? We can simply return -1, like any other failures

Reply:
also why the fprintf, you can simply use tpp_log() like the mutex_init function case